app/src/main/java/sdumchykov/androidApp/presentation/viewPager/contacts/MyContactsFragment.kt

```diff
@@ -0,0 +1,302 @@
+import dagger.hilt.android.AndroidEntryPoint
+import sdumchykov.androidApp.databinding.AddContactsDialogBinding
+import sdumchykov.androidApp.databinding.FragmentMyContactsBinding
+import sdumchykov.androidApp.domain.model.UserModel
+import sdumchykov.androidApp.domain.utils.Constants.HARDCODED_IMAGE_URL
+import sdumchykov.androidApp.presentation.base.BaseFragment
+import sdumchykov.androidApp.presentation.utils.SwipeToDeleteCallback
+import sdumchykov.androidApp.presentation.viewPager.ViewPagerFragment
+import sdumchykov.androidApp.presentation.viewPager.ViewPagerFragmentDirections
+import sdumchykov.androidApp.presentation.viewPager.contacts.adapter.UsersAdapter
+import sdumchykov.androidApp.presentation.viewPager.contacts.adapter.listener.UsersListener
+
+private const val SNACKBAR_TIME_LENGTH = 5000
+
+@AndroidEntryPoint
+class MyContactsFragment :
+    BaseFragment<FragmentMyContactsBinding>(FragmentMyContactsBinding::inflate) {
+
+    private val pagerFragment by lazy { parentFragment as ViewPagerFragment }
+    private val parentViewModel by lazy { pagerFragment.myContactsViewModel }
+    private val usersAdapter: UsersAdapter by lazy {
+        UsersAdapter(
+            usersListener = object : UsersListener {
+                override fun onUserClickAction(userModel: UserModel, adapterPosition: Int) {
+                    Log.d("MainActivity", "onUserClickAction: ${userModel.id}")
+                    val action =
+                        ViewPagerFragmentDirections.actionViewPagerFragmentToContactProfileFragment(
+                            userModel
+                        )
+                    Navigation.findNavController(binding.root).navigate(action)
+                }
+
+                override fun onTrashIconClickAction(userModel: UserModel, userIndex: Int) {
+                    deleteContact(userModel, userIndex)
+                }
+
+                override fun onContactRemove(userModel: UserModel) {
+                    parentViewModel.removeItem(userModel)
+                }
+
+                override fun onMultiselectActivated() {
+                    usersAdapter.enableMultiselect()
+                    enableMultiselectViewState()
+                    parentViewModel.selectedEvent.value = true
+                    disableContactSwipe()
+                    initRecyclerView()
+                    imageViewMyContactsCancelSetOnClickListener()
+                }
+
+                override fun onContactSelectedStateChanged() {
+                    if (usersAdapter.areAllItemsUnselected()) {
+                        disableMultiselectViewState()
+                        parentViewModel.selectedEvent.value = false
+                        enableContactSwipe()
+                        initRecyclerView()
+                    }
+                }
+
+                // TODO я хочу цю функцію винести безпосередньо у клас, а не тримати її у анонімному об'єкті, але тоді нема доступу до onContactSelectedStateChanged(). Як жити?
```

Comment:
Suicide is not a solution!

Reply:
Did you try to access it like yourListener.desiredMethod?
Also, you can implement the listener in you activity but then all your methods will be in the class